### ISSUE TITLE
(default) Invalid model for QF_NRA formula (src/theory/arith/nl/nl_model.cpp:1069, 366)

### ISSUE THREAD

**Issue by rainoftime** [CVC4/CVC4#5358]
Hi, for the following formula,
cvc4 e4567666de0c
~~~~
(set-logic QF_NRA)
(declare-fun _substvar_14_ () Real)
(declare-fun _substvar_15_ () Real)
(declare-fun r0 () Real)
(declare-fun r1 () Real)
(declare-fun r2 () Real)
(declare-fun r6 () Real)
(assert (= 33.0 (* r1 r1 r2 r0) _substvar_15_ r6 _substvar_14_))
(check-sat)
(assert (<= 0.0 r6 r6 r1))
(check-sat)
~~~~

~~~~
$ cvc4 -i --check-models xx.smt2 
sat
Fatal failure within void CVC4::smt::CheckModels::checkModel(CVC4::smt::Model*, CVC4::context::CDList<CVC4::NodeTemplate<true> >*, bool) at /home/peisen/test/tofuzz/CVC4/src/smt/check_models.cpp:258
Internal error detectedSmtEngine::checkModel(): ERRORS SATISFYING ASSERTIONS WITH MODEL:
assertion:     (and (<= (/ 0 1) r6) (<= r6 r6) (<= r6 r1))
simplifies to: false
expected `true'.
Run with `--check-models -v' for additional diagnostics.
Aborted
~~~~

**Comment by rainoftime** [comment]
src/theory/arith/nl/nl_model.cpp:366
~~~~
(set-logic QF_NRA)
(declare-fun _substvar_47_ () Bool)
(declare-fun _substvar_53_ () Real)
(declare-fun v0 () Bool)
(declare-fun v4 () Bool)
(declare-fun r1 () Real)
(declare-fun r3 () Real)
(declare-fun r5 () Real)
(declare-fun r8 () Real)
(declare-fun v49 () Bool)
(assert (or (or v0 v4) v49 _substvar_47_))
(assert (or (= 0.0 r3 r8 (/ r1 r8) (+ 5.0 0.91 0.91 r5 71418009583.0) 0.91) v49 (= (/ 0.4 r1) r3 r8 0.0 _substvar_53_ 0.91)))
(check-sat)
~~~~
~~~~
$ cvc4 --check-models xx.smt2
Fatal failure within bool CVC4::theory::arith::nl::NlModel::addCheckModelSubstitution(CVC4::TNode, CVC4::TNode) at CVC4/src/theory/arith/nl/nl_model.cpp:366
Check failure

 d_check_model_witnesses.find(v) == d_check_model_witnesses.end()
We tried to add a substitution where we already had a witness term.

Aborted
~~~~

**Comment by nafur** [comment]
First instance is fixed by e2aff722e0b1072e90bd0c77e7030957364283cc, second and third persist on current master.

**Comment by ajreynol** [comment]
I can't reproduce this on master.  This may have been fixed by recent simplifications to our use of witness terms.
